ginga/util/plots.py

```diff
@@ -132,6 +135,100 @@ def plot(self, xarr, yarr, xtitle=None, ytitle=None, title=None,
     def get_data(self):
         return self.fig, self.xdata, self.ydata
 
+    def autoscale(self, axis):
+        self.ax.autoscale(enable=True, axis=axis)
+        self.draw()
+        self.ax.autoscale(enable=False, axis=axis)
+
+        x1, x2 = self.ax.get_xlim()
+        y1, y2 = self.ax.get_ylim()
+        self.make_callback('limits-set', dict(x_lim=(x1, x2), y_lim=(y1, y2)))
+
+    def connect_ui(self):
+        canvas = self.fig.canvas
+        connect = canvas.mpl_connect
+        # This one is not ready for prime time...
+        # connect("motion_notify_event", self.plot_motion_notify)
+        connect("button_press_event", self.plot_button_press)
+        connect("scroll_event", self.plot_scroll)
+
+    def enable(self, **kwargs):
+        self.can.update(kwargs)
+
+    def plot_scroll(self, event):
+        if not self.can.zoom:
+            return
+        # Matplotlib only gives us the number of steps of the scroll,
+        # positive for up and negative for down.
+        if event.step > 0:
+            delta = 0.9
+            # self.plot_zoomlevel += 1.0
+        elif event.step < 0:
+            delta = 1.1
+            # self.plot_zoomlevel -= 1.0
+
+        #self.plot_panzoom()
+
+        x1, x2 = self.ax.get_xlim()
+        xrng = x2 - x1
+        xinc = (delta * xrng - xrng) / 2.0
```

Comment:
I think `* 0.5` is a bit faster than `/ 2.0` but that's probably premature optimization, so I'll leave it up to you.

Reply:
fixed, see update to be pushed.